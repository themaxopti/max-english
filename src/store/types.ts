import { InitialStateRecucer } from '../modules/testModule/state/test.reucer'
import { SomeInitialState, SOME_NAME } from './some.slice'
import { AnyAction, CombinedState } from '@reduxjs/toolkit'
import { Reducer } from 'react'
import { NAME_TEST } from '../modules/testModule/state/test.reucer'

export type Reducers = {
  [SOME_NAME]: SomeInitialState
  [NAME_TEST]: InitialStateRecucer
}
