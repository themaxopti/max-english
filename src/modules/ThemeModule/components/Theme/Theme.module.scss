.circle {
    width: 44px;
    height: 44px;
    border-radius: 100%;
    background-color: red;
    cursor: pointer;
}

.circle__navigation {
    position: relative;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.circle__language__content {
    position: absolute;
    padding: 1rem;
    background-color: #EEEEEE;
    top: 60px;
    width: 100px;
    border-radius: 4px;
    display: flex;
    flex-direction: column;
    row-gap: 20px;
    align-items: center;
}