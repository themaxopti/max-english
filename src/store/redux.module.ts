import { SomeInitialState, SOME_NAME } from './some.slice'
import { persistReducer, persistStore } from 'redux-persist'
import {
  AnyAction,
  CombinedState,
  configureStore,
  EmptyObject,
  Reducer,
  ReducersMapObject,
  Slice,
  ThunkMiddleware
} from '@reduxjs/toolkit'
import { EnhancedStore } from '@reduxjs/toolkit/dist/configureStore'
import { Module } from '../core/coreModule'
import { RootState } from './store'
import { Reducers } from './types'
import thunk from 'redux-thunk'

export type GetStateType = Reducer<CombinedState<Reducers>, AnyAction>

export class ReduxModule<T extends Reducer<any, AnyAction> | ReducersMapObject<any, AnyAction>> {
  public store
  constructor(rootReducer: T) {
    this.store = configureStore({ reducer: rootReducer, middleware: [thunk] })
  }
}
