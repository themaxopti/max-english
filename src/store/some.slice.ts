import { createSlice } from "@reduxjs/toolkit";
import React from "react";


const slice2 = createSlice({
    name: 'ops',
    initialState: { some: 0 },
    reducers: {
    },
})

export const someSlice2 = slice2



// const slice3 = createSlice({
//     name: 'bops',
//     initialState: { some: 0, ops: '10' },
//     reducers: {
//     },
// })

// export const someSlice3 = slice3