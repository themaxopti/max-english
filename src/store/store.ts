import { languageReducer } from '../modules/LanguageModule/state/reducers/language.slice'
import { combineReducers, configureStore } from '@reduxjs/toolkit'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { ReduxModule } from './redux.module'
import { Reducers } from './types'
import { someSlice } from '../modules/testModule/state/test.reucer'
import { someSlice2 } from './some.slice'
import { themeReducer } from '../modules/ThemeModule/state/reducers/theme.slice'
import storage from 'redux-persist/lib/storage'
import { persistReducer } from 'redux-persist'
import { persistStore } from 'redux-persist'
import thunk from 'redux-thunk'

const rootReducer = combineReducers({
  // learningNavbar: learningNavbarReducer,
  theme: themeReducer,
  language: languageReducer
})

const persistConfig = {
  key: 'root',
  storage
  // blackList: ['theme']
}
const persistedReducer = persistReducer(persistConfig, rootReducer)

// export type RootReducer = typeof rootReducer
const store = configureStore({ reducer: persistedReducer, middleware: [thunk] })

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const persistor = persistStore(store)
export default store
