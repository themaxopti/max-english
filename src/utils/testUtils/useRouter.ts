export function testRouter(useRouter: any, query: any, push: any) {
  useRouter.mockReturnValue({
    query,
    push
  })
}
