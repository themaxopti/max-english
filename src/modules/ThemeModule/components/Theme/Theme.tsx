import React from 'react'
import { motion } from 'framer-motion'
import { useState, useCallback } from 'react'
import s from './Theme.module.scss'
import { useAppDispatch } from '../../../../store/store'
import { changeMainTheme, Titles } from '../../state/reducers/theme.slice'
import { useSelector } from 'react-redux'
import { mainTheme, themesselector, themesSelector } from '../../state/selectors/theme.selectors'

export const Theme = React.memo(() => {
  const [active, setActive] = useState(false)
  // const [currentColor, setCurrentColor] = useState('#2C80FF')
  const colors = useSelector(themesselector)
  const currentColor = useSelector(mainTheme).primaryColor
  const dispatch = useAppDispatch()

  return (
    <>
      <motion.nav
        tabIndex={0}
        onBlur={() => {
          setActive(false)
        }}
        onClick={() => setActive(!active)}
        data-testid='theme'
        initial={false}
        animate={active ? 'open' : 'closed'}
        className={s.circle__navigation}
      >
        <motion.div
          whileTap={{ scale: 1.2 }}
          whileHover={{ scale: 1.1 }}
          className={s.circle}
          data-testid='circle'
          style={{
            background: currentColor
          }}
        ></motion.div>

        <motion.div
          data-testid='theme__burger'
          variants={{
            open: {
              clipPath: 'inset(0% 0% 0% 0% round 10px)',
              transition: {
                type: 'spring',
                bounce: 0,
                duration: 0.7,
                delayChildren: 0.3,
                staggerChildren: 0.05
              }
            },
            closed: {
              clipPath: 'inset(10% 50% 90% 50% round 10px)',
              transition: {
                type: 'spring',
                bounce: 0,
                duration: 0.3
              }
            }
          }}
          className={s.circle__language__content}
        >
          {colors.map((el, i) => (
            <motion.div
              className={s.circle}
              data-testid={`circleEl`}
              onClick={() => {
                // setCurrentColor(el.color)
                // @ts-ignore
                dispatch(changeMainTheme(el.title))
              }}
              whileHover={{ scale: 1.1 }}
              key={i}
              style={{
                background: el.color
              }}
            >
              {/* {el} */}
            </motion.div>
          ))}
        </motion.div>
      </motion.nav>
    </>
  )
})
