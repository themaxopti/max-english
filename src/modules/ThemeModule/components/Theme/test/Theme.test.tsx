import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from 'store/store'
import { createMockRouter } from '../../../../../utils/testUtils/createMockRouter'
import { Theme } from '../Theme'

describe('Test Theme', () => {
  jest.useFakeTimers()

  test('Should open theme burger', async () => {
    const view = render(
      <Provider store={store}>
        <Theme />
      </Provider>
    )

    const theme = screen.getByTestId('theme')
    const burger = screen.getByTestId('theme__burger')
    expect(burger).toHaveStyle('clip-path: inset(10% 50% 90% 50% round 10px);')
    fireEvent.click(theme)

    await waitFor(
      () => {
        expect(screen.getByTestId('theme__burger')).toHaveStyle(
          'clip-path: inset(0% 0% 0% 0% round 10px);'
        )
        // expect(screen.getAllByTestId('circle')).
      },
      { timeout: 50000 }
    )

    const burgerElement = screen.getAllByTestId(`circleEl`)
    burgerElement.forEach(el => {
      fireEvent.click(el)
      const mainCircle = screen.getByTestId(`circle`)
      console.log(el.style.background)
      expect(mainCircle).toHaveStyle(`background:${el.style.background}`)
      expect(mainCircle).not.toHaveStyle(`background:red`)
    })
    expect(view).toMatchSnapshot()
  })
})
