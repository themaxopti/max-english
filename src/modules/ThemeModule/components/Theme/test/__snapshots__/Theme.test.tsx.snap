// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Theme Should open theme burger 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="circle__navigation"
        data-testid="theme"
        tabindex="0"
      >
        <div
          class="circle"
          data-testid="circle"
          style="background: rgb(108, 255, 240);"
        />
        <div
          class="circle__language__content"
          data-testid="theme__burger"
          style="clip-path: inset(0% 0% 0% 0% round 10px);"
        >
          <div
            class="circle"
            data-testid="circleEl"
            style="background: rgb(44, 128, 255);"
          />
          <div
            class="circle"
            data-testid="circleEl"
            style="background: rgb(220, 45, 54);"
          />
          <div
            class="circle"
            data-testid="circleEl"
            style="background: rgb(108, 255, 240);"
          />
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="circle__navigation"
      data-testid="theme"
      tabindex="0"
    >
      <div
        class="circle"
        data-testid="circle"
        style="background: rgb(108, 255, 240);"
      />
      <div
        class="circle__language__content"
        data-testid="theme__burger"
        style="clip-path: inset(0% 0% 0% 0% round 10px);"
      >
        <div
          class="circle"
          data-testid="circleEl"
          style="background: rgb(44, 128, 255);"
        />
        <div
          class="circle"
          data-testid="circleEl"
          style="background: rgb(220, 45, 54);"
        />
        <div
          class="circle"
          data-testid="circleEl"
          style="background: rgb(108, 255, 240);"
        />
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
